exasol_transformers_extension/udfs/question_answering_udf.py

```diff
@@ -41,6 +39,16 @@ def run(self, ctx):
 
         self.clear_device_memory()
 
+    def _set_device(self, device_id: Optional[int]) -> None:
+        """
+        Set device to push models
+
+        :param device_id: Either the id of cuda device or None implying CPU
+        """
+        device_name = f"cuda:{device_id}" \
+            if torch.cuda.is_available() and device_id is not None else "cpu"
+        self.device = torch.device(device_name)
+
```

Comment:
Extract method completely from classes, such that you can test it without running the UDF

Reply:
I thought this method will be the part of base class that we will handle in the ticket #19 
When refactoring the class hierarchy, we will not need to call UDF to test the base class